WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderPostDetailHeaderView.java

```diff
@@ -0,0 +1,215 @@
+package org.wordpress.android.ui.reader.views;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+import org.wordpress.android.datasets.ReaderBlogTable;
+import org.wordpress.android.models.ReaderBlog;
+import org.wordpress.android.models.ReaderPost;
+import org.wordpress.android.ui.reader.ReaderActivityLauncher;
+import org.wordpress.android.ui.reader.actions.ReaderActions;
+import org.wordpress.android.ui.reader.actions.ReaderBlogActions;
+import org.wordpress.android.ui.reader.utils.ReaderUtils;
+import org.wordpress.android.util.GravatarUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.android.util.PhotonUtils;
+import org.wordpress.android.util.ToastUtils;
+import org.wordpress.android.widgets.WPNetworkImageView;
+
+/**
+ * topmost view in post detail - shows blavatar + avatar, author name, blog name, and follow button
+ */
+public class ReaderPostDetailHeaderView extends LinearLayout {
+
+    private ReaderPost mPost;
+    private ReaderFollowButton mFollowButton;
+
+    public ReaderPostDetailHeaderView(Context context) {
+        super(context);
+        initView(context);
+    }
+
+    public ReaderPostDetailHeaderView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        initView(context);
+    }
+
+    public ReaderPostDetailHeaderView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        initView(context);
+    }
+
+    private void initView(Context context) {
+        View view = inflate(context, R.layout.reader_post_detail_header_view, this);
+        mFollowButton = (ReaderFollowButton) view.findViewById(R.id.header_follow_button);
+    }
+
+    public void setPost(@NonNull ReaderPost post) {
+        mPost = post;
+
+        TextView txtBlogName = (TextView) findViewById(R.id.text_header_blog_name);
+        TextView txtAuthorName = (TextView) findViewById(R.id.text_header_author_name);
+
+        boolean hasBlogName = mPost.hasBlogName();
+        boolean hasAuthorName = mPost.hasAuthorName();
+
+        if (hasBlogName && hasAuthorName) {
+            txtBlogName.setText(mPost.getBlogName());
+            // don't show author name if it's the same as the blog name
+            if (mPost.getAuthorName().equals(mPost.getBlogName())) {
+                txtAuthorName.setVisibility(View.GONE);
```

Comment:
This is just a suggestion, but when author and blog name are same, maybe we could show ```txtAuthorName``` instead of ```txtBlogName``` which is bolder and more prominent. What do you think?